### PR TITLE
Feature/elastic search

### PR DESCRIPTION
**Pull request by melsawy**
The branch include implementation for issues #5157, #5047 and #5149 which related to Pender as annotation, editing title/description, claims extensions to Media
Implementation details:
 - Dropped `data` column from media and account and added Pender data as a new annotations called `embed`
 - Migrate existing data to annotations
 - Dropped `title` and `description` columns from media 
 - Made `Media.url` optional
 - Allow user to update Pender data through `information` method that accept a json format. 
	*Example `info = {title: 'new title', description: 'new description'}`
 - Modify `jsondata` to be updated with overridden keys.
 - Added `Embed.quote` attribute to support quotes through `information` method that accept a json format. 
	*Example to add claims extension `info = {title: 'new title', description: 'new description', quote: 'Media quote'}`

### CODE REVIEW

File: app/controllers/api/v1/search_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  module V1
+    class SearchController < Api::V1::BaseApiController
+      include SearchDoc
+
+      skip_before_filter :authenticate_from_token!
+
+      def create
+        query_string = params[:query]
+        repository = Elasticsearch::Persistence::Repository.new
+        repository.index = 'checkdesk_application_development_annotations'
```
Comment (caiosba) [inline app/controllers/api/v1/search_controller.rb]: The index name should not be hard-coded... try to read this property: https://github.com/meedan/check-api/blob/develop/app/models/concerns/annotation_base.rb#L66

File: app/controllers/api/v1/search_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        repository.type = 'annotation'
+        result = repository.search(query: { query_string: {query: query_string}})
```
Comment (caiosba) [inline app/controllers/api/v1/search_controller.rb]: Do you need to re-instantiate the repository? I was hoping that you could use `Annotation.search`, as we do on annotation_base

File: db/migrate/20160927085541_remove_data_from_accounts.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class RemoveDataFromAccounts < ActiveRecord::Migration
+  def change
+    pender = Bot.where(name: 'Pender').last
+    Account.find_each do |account|
+      em = Embed.new
+      em.embed = account.data
+      em.annotated = account
+      em.annotator = pender unless pender.nil?
+      em.save
```
Comment (caiosba) [inline db/migrate/20160927085541_remove_data_from_accounts.rb]: Please replace this `save` by `save!`

File: db/migrate/20161005065727_remove_title_and_description_from_medias.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class RemoveTitleAndDescriptionFromMedias < ActiveRecord::Migration
+  def change
+    remove_column :medias, :title, :string
+    remove_column :medias, :description, :string
```
Comment (caiosba) [inline db/migrate/20161005065727_remove_title_and_description_from_medias.rb]: `remove_column` doesn't take a third argument

File: test/controllers/graphql_controller_test.rb [hunk 5]
```diff
@@ -105,11 +106,23 @@ def setup
...
+    p = create_project team: @team
+    m = create_valid_media project_id: p.id
+    query = "query GetById { media(id: \"#{m.id}\") { jsondata(context: \"#{p}\") } }"
+    post :create, query: query
+    assert_response :success
+    # calling without context
+    query = "query GetById { media(id: \"#{m.id}\") { jsondata(context: nil) } }"
+    post :create, query: query
+    assert_response :success
+  end
```
Comment (caiosba) [inline test/controllers/graphql_controller_test.rb]: I was expecting something different for this test. Two medias are created, for two different projects. One query for each project, then we assert that only the relevant media was returned. Then a third case for when a context is not provided. Also, the client doesn't send it as `nil`, but as an empty string.

File: test/controllers/search_controller_test.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+class SearchControllerTest < ActionController::TestCase
+  def setup
+    super
+    @controller = Api::V1::SearchController.new
+  end
+
+  test "should get create" do
+    #get :create
+    #assert_response :success
```
Comment (caiosba) [inline test/controllers/search_controller_test.rb]: Why this test is missing?

File: test/models/bot_test.rb [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  test "should create bot" do
+    assert_difference 'Bot.count' do
+      create_bot
+    end
+  end
+
+  test "should not save bot without name" do
+    project = Project.new
+    assert_not project.save
+  end
```
Comment (caiosba) [inline test/models/bot_test.rb]: Should be "bot" instead of "project" here :)

File: app/controllers/api/v1/search_controller.rb (hunk 2 above)
Comment (melsawy) [inline app/controllers/api/v1/search_controller.rb]: no will use Annotation.search

File: db/migrate/20161005065727_remove_title_and_description_from_medias.rb (hunk 4 above)
Comment (melsawy) [inline db/migrate/20161005065727_remove_title_and_description_from_medias.rb]: Done

File: test/controllers/search_controller_test.rb (hunk 6 above)
Comment (melsawy) [inline test/controllers/search_controller_test.rb]: Will add this test when back to issue #4830

File: test/models/bot_test.rb (hunk 7 above)
Comment (melsawy) [inline test/models/bot_test.rb]: Done

**Pull request closed without merging**